src/main/java/it/gov/pagopa/reward/connector/repository/UserInitiativeCountersAtomicOpsRepositoryImpl.java

```diff
@@ -109,4 +111,16 @@ public Flux<UserInitiativeCounters> findByInitiativesWithBatch(String initiative
         return mongoTemplate.find(query, UserInitiativeCounters.class);
     }
 
+    @Override
+    public Mono<UserInitiativeCounters> unlockPendingTrx(String trxId) {
+        return mongoTemplate
+                .findAndModify(
+                        Query.query(Criteria.where(PENDING_TRX_ID_FIELD).is(trxId)),
```

Comment:
you have to search for the involved counter by id having the pendingTrx.id requested
Which is a combination of initiativeId and entityId (userId or familyId based on initiative target type)
see: it.gov.pagopa.reward.service.synchronous.op.BaseTrxSynchronousOp#retrieveCounterEntityId(it.gov.pagopa.reward.dto.InitiativeConfig, it.gov.pagopa.reward.model.OnboardingInfo, it.gov.pagopa.reward.dto.trx.TransactionDTO)


Actually there is not an index on the field you are using to search, building the counterId could avoid it